internal/arcade/client_test.go

```diff
@@ -105,6 +106,37 @@ var _ = Describe("Client", func() {
 			})
 		})
 
+		When("provider is rancher and the 60 second short expiry has passed", func() {
+			BeforeEach(func() {
+				provider = "rancher"
+				client = NewClient(server.URL())
+				client.WithAPIKey("test-api-key")
+				client.WithShortExpiration(2)
+				server.AppendHandlers(ghttp.CombineHandlers(
+					ghttp.VerifyHeaderKV("api-key", "test-api-key"),
+					ghttp.VerifyRequest(http.MethodGet, "/tokens", "provider=rancher"),
+					ghttp.RespondWith(http.StatusOK, `{"token":"some.bearer.token"}`),
+				))
+
+				// call to test the short expiration
+				server.AppendHandlers(ghttp.CombineHandlers(
+					ghttp.VerifyHeaderKV("api-key", "test-api-key"),
+					ghttp.VerifyRequest(http.MethodGet, "/tokens", "provider=rancher"),
+					ghttp.RespondWith(http.StatusOK, `{"token":"new.bearer.token"}`),
+				))
+			})
+			It("returns a new token", func() {
+				Expect(err).To(BeNil())
+				Expect(token).To(Equal("some.bearer.token"))
+
+				time.Sleep(3 * time.Second)
```

Comment:
I'm not sure how to best test this since we mock the responses and the logic check is internal to arcade the sleep here doesn't really do anything but not sure if there's a better way to test what I'm trying to test. Open to suggestions.

Reply:
Seems like after you modify the interface, you might need to regenerate the fakes. This might solve the test failing. I will take a deeper look by running the code locally

Reply:
Yeah I had run it but it wasn't actually updating the fakes. I ended up updating to the latest version of go and counterfeiter and updating the needed packages. But that's a separate issue than my comment above, thinking about it I don't think it can be tested here - there is a proper test from within `arcade` but I think keeping the test

Reply:
I only find one suggestion. Also, I was not able to see where the new value is being used. I see it's declared and initialize. I wonder if it's sent to arcade and arcade is the service that actually use that value?

Reply:
Yeah, arcade is the one that reads it and does the [magic](https://github.com/homedepot/arcade/pull/11/files#diff-fe11b69a179e51956cdc1c2ebe606e9ddf75098bf18107359f56b93062d82e28R31).